l1/queue.go

```diff
@@ -0,0 +1,63 @@
+package l1
+
+import (
+	"github.com/NethermindEth/juno/l1/internal/contract"
+)
+
+// queue is a priority queue that implements a "confirmation window."
+// Given an L1 height, all [contract.LogStateUpdate]s emitted at least
+// confirmationPeriod blocks before the height are "confirmed".
+//
+// Its abnormal API reflects the fact that it is only meant to be used
+// by the L1 verifier.
+type queue struct {
```

Comment:
This queue doesn't ensure that there are no missing blocks, which can happen if subscriptions fail and new blocks were created during the creation of a new subscription.

Reply:
Right. You think we should add a comment about that somewhere?

Reply:
I realised it doesn't matter if there are missing blocks all we care about is the difference between the block number and the confirmation period.